### PR TITLE
 IA-3394 Feat: improve profile api performance

### PR DESCRIPTION
**Pull request by madewulf**
Feat: improve profile api performance, by removing useless data and prefetching useful data

Related JIRA tickets : IA-3394

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [x] Are there enough tests

### CODE REVIEW

File: iaso/models/base.py [hunk 1]
```diff
@@ -1358,27 +1370,50 @@ def as_dict(self):
...
-            "language": self.language,
-            "user_id": self.user.id,
-            "dhis2_id": self.dhis2_id,
-            "home_page": self.home_page,
-            "phone_number": self.phone_number.as_e164 if self.phone_number else None,
-            "country_code": region_code_for_number(self.phone_number).lower() if self.phone_number else None,
-            "projects": [p.as_dict() for p in self.projects.all().order_by("name")],
-        }
+        if not small:
+            print("not small")
```
Comment (hakifran) [inline iaso/models/base.py]: I think this print should be removed

File: iaso/models/org_unit.py [hunk 2]
```diff
@@ -509,6 +509,17 @@ def as_small_dict(self):
             res["search_index"] = self.search_index
         return res
 
+    def as_very_small_dict(self):
+        res = {
```
Comment (hakifran) [inline iaso/models/org_unit.py]: Not blocking but I think this dict should be return directly without passing it to res before to return it

**Review by hakifran — CHANGES_REQUESTED** [review]
I don't know if it's only on my local machine but these changes crushed test_upload_valid_csv and test_upload_valid_csv_with_perms test.

**Comment by madewulf** [comment]
@hakifran there are changes to the test files https://github.com/BLSQ/iaso/pull/1607/files#diff-bc4da97802d782f1e9e50fc072a80ed8b252157a2e01b866130dfada1634e964 but that should not delete the files on your machine

**Review by hakifran — APPROVED** [review]

**Pull request merged**